lib/Importer/pytorch/PyTorchModelLoader.cpp

```diff
@@ -0,0 +1,115 @@
+ */
+
+#include "PyTorchModelLoader.h"
+
+namespace {
+glow::Type ptTypeToGlowType(const at::CompleteTensorType &ptType) {
+  // TODO: get correct ElemKind
+  std::vector<size_t> dims;
+  for (auto &size : ptType.sizes()) {
+    dims.push_back(size);
+  }
+  return glow::Type(glow::ElemKind::FloatTy, dims);
+}
+} // namespace
+
+glow::Placeholder *PyTorchModelLoader::loadValue(const torch::jit::Value *val) {
+  // TODO: do we need to care about optional IValues here?
+  assert(val->isCompleteTensor());
+  auto ptType = val->type()->cast<at::CompleteTensorType>();
+  auto glowType = ptTypeToGlowType(*ptType.get());
+  glow::Placeholder *ph =
+      mod_.createPlaceholder(&glowType, "input", /*isTrainable*/ false);
+  valueMap_[val] = ph->getOutput();
+  return ph;
+}
+
+void PyTorchModelLoader::loadNode(const torch::jit::Node *ptNode) {
+  glow::Node *glowNode = nullptr;
+
+  auto inputs = ptNode->inputs();
+  auto outputs = ptNode->outputs();
+
+  switch (ptNode->kind()) {
+  case at::aten::mul: {
+    assert(inputs.size() == 2);
+    glow::NodeValue rhs = valueMap_.at(inputs[0]);
+    glow::NodeValue lhs = valueMap_.at(inputs[1]);
+    glowNode = loadRes_->f->createMul("mul", rhs, lhs);
+    valueMap_[outputs[0]] = glowNode->getNthResult(glow::MulNode::ResultIdx);
+    break;
+  }
+  case at::aten::div: {
+    assert(inputs.size() == 2);
+    glow::NodeValue rhs = valueMap_.at(inputs[0]);
+    glow::NodeValue lhs = valueMap_.at(inputs[1]);
+    glowNode = loadRes_->f->createDiv("div", rhs, lhs);
+    valueMap_[outputs[0]] = glowNode->getNthResult(glow::DivNode::ResultIdx);
+    break;
+  }
+  default:
+    assert(false && "unrecognized node");
+  }
+}
+
+std::unique_ptr<ModelLoaderResult>
+PyTorchModelLoader::load(at::ArrayRef<torch::jit::IValue> &inputs) {
+  loadRes_.reset(new ModelLoaderResult());
+  static int32_t nextFuncId = 0;
+  loadRes_->f =
+      mod_.createFunction(glow::strFormat("PyTorchFunction_%d", nextFuncId++));
+
+  auto subgraphInputValues = subgraph_->inputs();
+
+  assert(inputs.size() == subgraphInputValues.size());
+
+  for (size_t i = 0; i < subgraphInputValues.size(); ++i) {
+    torch::jit::Value *inputValue = subgraphInputValues[i];
+    const c10::IValue inputIValue = inputs[i];
+    inputValue->inferTypeFrom(inputIValue.toTensor());
+    loadRes_->inputPlaceholders.push_back(loadValue(inputValue));
+  }
+
+  // Nodes are topologically sorted.
+  for (auto node : subgraph_->nodes()) {
+    loadNode(node);
+  }
+
+  for (torch::jit::Value *output : subgraph_->outputs()) {
+    glow::NodeValue outputNodeValue = valueMap_.at(output);
+    auto *save = loadRes_->f->createSave("save", outputNodeValue);
+    loadRes_->outputPlaceholders.push_back(save->getPlaceholder());
+  }
+
+  return std::move(loadRes_);
+}
+
+PyTorchModelLoader::PyTorchModelLoader(glow::Module &mod,
+                                       torch::jit::Graph *subgraph)
+    : mod_(mod), subgraph_(subgraph) {}
+
+// static
```

Comment:
shall we have a static list for this?

Reply:
I don't think so, eventually this should probably look more like onnxifi's checkGraphCompatibility

Reply:
> I don't think so, eventually this should probably look more like onnxifi's checkGraphCompatibility

Check my comments lol